dynparquet/schema.go

```diff
@@ -1231,27 +1235,38 @@ func (s *Schema) NewWriter(w io.Writer, dynamicColumns map[string][]string) (*pa
 		)
 	}
 
-	return parquet.NewGenericWriter[any](w,
+	writerOptions := []parquet.WriterOption{
 		ps.Schema,
 		parquet.ColumnIndexSizeLimit(ColumnIndexSize),
 		parquet.BloomFilters(bloomFilterColumns...),
 		parquet.KeyValueMetadata(
 			DynamicColumnsKey,
 			serializeDynamicColumns(dynamicColumns),
 		),
-		parquet.SortingWriterConfig(
-			parquet.SortingColumns(cols...),
-		),
-	), nil
+	}
+	sortingOptions := []parquet.SortingOption{parquet.SortingColumns(cols...)}
+	if s.uniquePrimaryIndex {
+		sortingOptions = append(sortingOptions, parquet.DropDuplicatedRows(true))
```

Comment:
Nit Is the conditional necessary? Could just be `parquet.DropDuplicatedRows(s.uniquePrimaryIndex)`